File: src/main/java/com/travel/domain/city/domain/log/CityLog.java
package com.travel.domain.city.domain.log;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
@Entity
@Table(name="cityLog")
public class CityLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long cityId;

    @Column
    private String userId;

    @CreatedDate
    private LocalDateTime createDate;

    @Builder
    public CityLog(Long cityId) {
        this.cityId = cityId;
    }

    public CityLog toEntity() {
        return CityLog.builder()
                .cityId(cityId)
                .build();
    }
}

File: src/main/kotlin/com/travel/domain/city/domain/log/CityLog.kt
package com.travel.domain.city.domain.log

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@EntityListeners(AuditingEntityListener::class)
@Entity
@Table(name = "cityLog")
class CityLog constructor(
        @Column
        val cityId: Long,

        @Column
        val userId: String,

        @CreatedDate
        val createDate: LocalDateTime,

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long? = null
) {
}
